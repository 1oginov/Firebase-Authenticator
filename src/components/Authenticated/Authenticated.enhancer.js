// @flow

import {
  compose, withHandlers, withProps, type HOC,
} from 'recompose';

import type { FirebaseApp } from '../../lib/firebaseApp';
import { redirect } from '../../lib/location';

type EnhancedComponentProps = {
  app: FirebaseApp,
  appInstance: Object,
};

const enhancer: HOC<*, EnhancedComponentProps> = compose(
  withHandlers({

    handleContinueClick: ({ app, appInstance }) => () => {
      const user = appInstance.auth().currentUser;

      if (!user) {
        throw new Error('User is missing');
      }

      const { refreshToken } = user;

      redirect(app.redirect.url.replace(app.redirect.refreshTokenPlaceholder, refreshToken));
    },

    handleSignOutClick: ({ appInstance }) => () => {
      appInstance.auth().signOut();
    },

  }),
  withProps(({ appInstance }) => {
    const user = appInstance.auth().currentUser;

    if (!user) {
      throw new Error('User is missing');
    }

    return {
      email: user.email || '',
      name: user.displayName || '',
      pictureUrl: user.photoURL || '',
    };
  }),
);

export default enhancer;
