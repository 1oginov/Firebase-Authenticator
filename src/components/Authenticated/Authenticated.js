// @flow

import * as React from 'react';

type Props = {
  email: string,
  handleContinueClick: () => void,
  handleSignOutClick: () => void,
  name: string,
  pictureUrl: string,
};

const Authenticated = ({
  email, handleContinueClick, handleSignOutClick, name, pictureUrl,
}: Props) => (
  <React.Fragment>

    <div>
      {'Signed in as '}
      <strong>{name}</strong>
      {' '}
      <em>{email}</em>
    </div>

    <div>
      <button onClick={handleContinueClick} type="button">Continue</button>
    </div>

    <div>
      <button onClick={handleSignOutClick} type="button">Sign out</button>
    </div>

    <div>
      <img alt="User" src={pictureUrl} />
    </div>

  </React.Fragment>
);

export default Authenticated;
