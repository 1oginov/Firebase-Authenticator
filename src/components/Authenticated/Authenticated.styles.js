// @flow

import { withStyles } from '@material-ui/core/styles';

const styles = {
  actions: {
    justifyContent: 'space-between',
  },
  media: {
    paddingTop: '100%',
  },
};

export default withStyles(styles);
