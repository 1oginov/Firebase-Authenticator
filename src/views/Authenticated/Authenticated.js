import firebase from 'firebase';
import * as React from 'react';

import * as C from '../../constants';
import { getQueryParameter, redirect } from '../../lib/location';

export default class Authenticated extends React.Component {
  state = {
    displayName: '',
    email: '',
  };

  componentDidMount() {
    const { displayName, email } = firebase.auth().currentUser;

    this.setState({ displayName, email });
  }

  handleContinue() { // eslint-disable-line class-methods-use-this
    const { refreshToken } = firebase.auth().currentUser;
    const redirectUrl = getQueryParameter(C.REDIRECT_QUERY_PARAMETER)
      .replace(C.REDIRECT_REFRESH_TOKEN_PLACEHOLDER, refreshToken);

    redirect(redirectUrl);
  }

  handleSignOut() { // eslint-disable-line class-methods-use-this
    firebase.auth().signOut();
  }

  render() {
    const { displayName, email } = this.state;

    return (
      <React.Fragment>

        <div>
          {'Signed in as '}
          <strong>{displayName}</strong>
          {' '}
          <em>{email}</em>
        </div>

        <div>
          <button onClick={this.handleContinue} type="button">Continue</button>
        </div>

        <div>
          <button onClick={this.handleSignOut} type="button">Sign out</button>
        </div>

      </React.Fragment>
    );
  }
}
