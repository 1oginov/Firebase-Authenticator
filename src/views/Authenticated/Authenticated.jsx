import firebase from 'firebase';
import * as React from 'react';

export default class Authenticated extends React.Component {
  state = {
    displayName: '',
    email: '',
  };

  componentDidMount() {
    const { displayName, email } = firebase.auth().currentUser;

    this.setState({ displayName, email });
  }

  handleContinue() { // eslint-disable-line class-methods-use-this
    const { refreshToken } = firebase.auth().currentUser;

    console.log(refreshToken);
  }

  handleSignOut() { // eslint-disable-line class-methods-use-this
    firebase.auth().signOut();
  }

  render() {
    const { displayName, email } = this.state;

    return (
      <React.Fragment>

        <div>
          {'Signed in as '}
          <strong>{displayName}</strong>
          {' '}
          <em>{email}</em>
        </div>

        <div>
          <button onClick={this.handleContinue} type="button">Continue</button>
        </div>

        <div>
          <button onClick={this.handleSignOut} type="button">Sign out</button>
        </div>

      </React.Fragment>
    );
  }
}
