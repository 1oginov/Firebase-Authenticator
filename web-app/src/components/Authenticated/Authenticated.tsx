import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import * as React from 'react';

type Props = {
  classes: { [string]: string },
  email: string,
  handleContinueClick: () => void,
  handleSignOutClick: () => void,
  name: string,
  pictureUrl: string,
};

const Authenticated = ({
  classes, email, handleContinueClick, handleSignOutClick, name, pictureUrl,
}: Props) => (
  <Card>

    {pictureUrl && (
      <CardMedia className={classes.media} image={pictureUrl} title="User" />
    )}

    <CardContent>

      <Typography gutterBottom={Boolean(name || email)} variant="h6">Signed in</Typography>

      {name && (
        <Typography>{name}</Typography>
      )}

      {email && (
        <Typography>{email}</Typography>
      )}

    </CardContent>

    <CardActions className={classes.actions}>
      <Button color="secondary" onClick={handleSignOutClick} size="small">Sign out</Button>
      <Button color="primary" onClick={handleContinueClick} size="small">Continue</Button>
    </CardActions>

  </Card>
);

export default Authenticated;
