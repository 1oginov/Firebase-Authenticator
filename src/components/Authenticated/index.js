// @flow

import Authenticated from './Authenticated';
import enhance from './Authenticated.enhancer';

export default enhance(Authenticated);
