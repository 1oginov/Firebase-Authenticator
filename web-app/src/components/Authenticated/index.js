// @flow

import Authenticated from './Authenticated';
import enhance from './Authenticated.enhancer';
import style from './Authenticated.styles';

export default style(enhance(Authenticated));
