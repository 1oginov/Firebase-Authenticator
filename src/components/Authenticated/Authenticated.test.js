// @flow

import classes from 'identity-obj-proxy';
import { shallow } from 'enzyme';
import React from 'react';

import Authenticated from './Authenticated';

describe('<Authenticated />', () => {
  it('renders without crashing and matches snapshot', () => {
    const wrapper = shallow(
      <Authenticated
        classes={classes}
        email="email@example.com"
        handleContinueClick={() => undefined}
        handleSignOutClick={() => undefined}
        name="Name"
        pictureUrl="https://example.com/picture.png"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
