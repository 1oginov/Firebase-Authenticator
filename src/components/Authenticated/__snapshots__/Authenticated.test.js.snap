// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticated /> renders without crashing and matches snapshot 1`] = `
<WithStyles(Card)>
  <WithStyles(CardMedia)
    className="media"
    image="https://example.com/picture.png"
    title="User"
  />
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      gutterBottom={true}
      variant="h6"
    >
      Signed in
    </WithStyles(Typography)>
    <WithStyles(Typography)>
      Name
    </WithStyles(Typography)>
    <WithStyles(Typography)>
      email@example.com
    </WithStyles(Typography)>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)
    className="actions"
  >
    <WithStyles(Button)
      color="secondary"
      onClick={[Function]}
      size="small"
    >
      Sign out
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
      size="small"
    >
      Continue
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
